### PR TITLE
Cache initial reads in `VortexFormat`

### PR DESCRIPTION
**Pull request by AdamGS**

### CODE REVIEW

File: vortex-datafusion/src/persistent/format.rs [hunk 1]
```diff
@@ -16,30 +16,55 @@ use datafusion_expr::Expr;
...
 use super::statistics::{array_to_col_statistics, uncompressed_col_size};
 use crate::can_be_pushed_down;
 
-#[derive(Debug, Default)]
+const DEFAULT_CACHE_SIZE: u64 = 256 * (2 << 20);
+
+#[derive(Debug)]
 pub struct VortexFormat {
     context: Arc<Context>,
+    initial_read_cache: Cache<StorePath, InitialRead>,
```
Comment (a10y) [inline vortex-datafusion/src/persistent/format.rs]: what's the lifetime of a FileFormat in Datafusion world? Is it tied to a SessionContext?

**Comment by robert3005** [comment]
Still need to pass it to VortexExec

File: vortex-datafusion/src/persistent/format.rs (hunk 1 above)
Comment (a10y) [reply vortex-datafusion/src/persistent/format.rs]: ok seems like DF does not management of the object lifetime.

Still though, is this the right layer to be doing caching? If the object is deleted and then overwritten, and user has no way to control caching, that feels bad.

Comment (gatesn) [reply vortex-datafusion/src/persistent/format.rs]: Yeah it's weird, I think it's either this or we move away from FileFormat entirely and go for a TableProvider. I've asked the folks over at DF what they think:  https://github.com/apache/datafusion/issues/13720

**Review by gatesn — APPROVED** [review]

**Pull request merged**